### PR TITLE
Add example app to see StateError

### PR DESCRIPTION
**Pull request by joshprzybyszewski-wf**
> Please apply the appropriate label to your PR:
> - bug / critical
> - improvement / optimization / tech-debt / UX
> - feature

### Description
Since the `StateError`s we throw aren't easy to see, let's just add an example app to be able to easily see them in the console

### Changes
Add another example app

### Semantic Versioning

- [x] **Patch**
  - [x] This change does not affect the public API
  - [ ] This change fixes existing incorrect behavior without any additions
- [ ] **Minor**
  - [ ] This change adds something to the public API
  - [ ] This change deprecates a part of the public API
* [ ] Major
  - [ ] This change modifies part of the public API in a backwards-incompatible manner
  - [ ] This change removes part of the public API

### Testing/QA

- [ ] CI passes

### Code Review

@dustinlessard-wf
@evanweible-wf
@jayudey-wf
@maxwellpeterson-wf
@sebastianmalysa-wf
@trentgrover-wf

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: example/errors/example.dart [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  MyManager() {
+    print('creating MyManager!');
+  }
+
+  @override
+  String get disposableTypeName => 'MyManager';
+
+  @override
+  Future<Null> onDispose() {
+    print('disposing MyManager');
```
Comment (georgelesica-wf) [inline example/errors/example.dart]: Maybe standardize on `ClassName.methodName` as a pattern?

**Comment by georgelesica-wf** [comment]
+1

**Comment by georgelesica-wf** [comment]
+1

**Comment by sebastianmalysa-wf** [comment]
QA +1 / +10

- ran the example and it works as expected and looks great :) 
- CI passes 

@Workiva/release-management-pp

**Pull request merged**